Library/Homebrew/cmd/--cache.rb

```diff
@@ -27,7 +27,8 @@ def self.__cache_args
       switch "--force-bottle",
              description: "Show the cache file used when pouring a bottle."
       flag "--bottle-tag=",
-           description: "Show the cache file used when pouring a bottle for the given tag."
+           description: "Show the cache file used when pouring a bottle for the given tag.",
+           hidden:      true
```

Comment:
Would prefer to keep `--bottle-tag`. Otherwise, doing things like
```
tag=big_sur; brew fetch --bottle-tag=$tag $formula && tar xf $(brew --cache --bottle-tag=$tag $formula)
```
becomes a lot messier.

Reply:
@carlocab Please find consensus with @reitermarkus on this. I'm just following the standard deprecation practises here.

Reply:
@carlocab alternatively: push to this PR reverting this bit and then ✅/merge.

Reply:
Fine with me!

Reply:
#15643